Dev 3.0

GigaZiv opened the pull request.

GigaZiv committed:
Dev 3.0

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/SearchAcitivty.kt:
```diff
@@ -49,6 +53,28 @@ class SearchActivity : AppCompatActivity() {
+    }
+
+    private fun init() {
+        val arrayTrack: ArrayList<Track> = arrayListOf()
```
Рекомендуется использовать котлиновские интерфейсы `MutableList` и `List` для работы со списками. Создать такие списки можно с помощью функций `mutableListOf()` и `listOf()` соответственно

GigaZiv replied:
В тех. задании нет таких требований. Так как можно создать 100000 способами. Как и выносить массив в ресурс. Так как я и сделал.

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/SearchAcitivty.kt:
```diff
@@ -49,6 +53,28 @@ class SearchActivity : AppCompatActivity() {
+        val arrayTrack: ArrayList<Track> = arrayListOf()
+        resources.getStringArray(R.array.tracks).forEach { s ->
+            val arr = s.split(",").toTypedArray()
+            arrayTrack.add(
+                Track(
+                    trackName = arr[0],
+                    artistName = arr[1],
+                    trackTime = arr[2],
+                    artworkUrl100 = arr[3]
+                )
```
Для добавления можно воспользоваться функцией `buildList` и в лямбде собрать спиок, получив на выходе немьютабельный список

GigaZiv replied:
Да, можно, знаю, но делал для демонстрации, так как для промышленного использования в этом нет никакого смысла.

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/SearchAcitivty.kt:
```diff
@@ -49,6 +53,28 @@ class SearchActivity : AppCompatActivity() {
+
+        val rw = findViewById<RecyclerView>(R.id.rcvSearch)
+        rw.layoutManager = LinearLayoutManager(this)
+        rw.adapter = SearchAdapter(arrayTrack)
```
В нейминге класса адаптера лучше отражать ту сущность, которую он отображает, т.е. более говорящим неймингом будет `TracksAdapter`

GigaZiv replied:
Хорошо, я переделаю

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/adapter/SearchAdapter.kt:
```diff
@@ -0,0 +1,63 @@
+        private val itemNameGroup: TextView by lazy { v.findViewById(R.id.twNameGroup) }
+        private val itemDuration: TextView by lazy { v.findViewById(R.id.twDuration) }
+
+        fun bind(i: Track) {
```
Нейминг аргумента тоже лучше сделать более говорящим, однобуквенные названия тяжело читать и использовать

GigaZiv replied:
Это сила привычки....

ArturNurtdinov reviewed (commented):
Размер плейсхолдера обложки и размер стрелки справа не соответствуют макетам. Все векторные ресурсы необходимо экспортировать с фигмы с их внутренними отступами:
![image](https://github.com/user-attachments/assets/94fc8a6a-3611-4404-8db3-724eadc43f7b)
Макет:
![image](https://github.com/user-attachments/assets/58472eb3-0785-4245-b3ff-dcacb88962f2)
Экспортировать нужно векторные ресурсы с внутренними отступами 45х45 или 24х24:
![image](https://github.com/user-attachments/assets/da87dbe3-49c8-48d3-b106-ed1cdb98aa58)

GigaZiv committed:
Dev 3.1

Исправлено:
1. Размер плейсхолдера обложки и размер стрелки справа не соответствуют макетам. Все векторные ресурсы необходимо экспортировать с фигмы с их внутренними отступами:

2. Рекомендуется использовать котлиновские интерфейсы MutableList и List для работы со списками. Создать такие списки можно с помощью функций mutableListOf() и listOf() соответственно

3. Нейминг аргумента тоже лучше сделать более говорящим, однобуквенные названия тяжело читать и использовать

4. В нейминге класса адаптера лучше отражать ту сущность, которую он отображает, т.е. более говорящим неймингом будет TracksAdapter
